cmake/modules/SetUpLinux.cmake

```diff
@@ -54,6 +54,32 @@ elseif(${SYSCTL_OUTPUT} MATCHES arm)
   else()
     message(FATAL_ERROR "There is no Setup for this compiler up to now. Don't know what to do. Stop cmake at this point.")
   endif()
+elseif(${SYSCTL_OUTPUT} MATCHES s390x)
+  message(STATUS "Found a 64bit system")
+  set(BIT_ENVIRONMENT "-m64")
+  set(SPECIAL_CINT_FLAGS "-DG__64BIT")
+  if(CMAKE_COMPILER_IS_GNUCXX)
+    message(STATUS "Found GNU compiler collection")
+    set(ROOT_ARCHITECTURE linuxs390xgcc)
+  elseif(CMAKE_CXX_COMPILER_ID STREQUAL Clang)
+    message(STATUS "Found CLANG compiler")
+    set(ROOT_ARCHITECTURE linuxs390xgcc)
+  else()
+    message(FATAL_ERROR "There is no Setup for this compiler up to now. Don't know what to do. Stop cmake at this point.")
+  endif()
+elseif(${SYSCTL_OUTPUT} MATCHES s390)
```

Comment:
Hi Mattias @ellert - the llvm people tell us at https://reviews.llvm.org/D33465 that this is wrong. We'll back this out if you don't object within the next couple of days... I hope you understand - if llvm explicitly says "wrong" it doesn't make much sense for us to overrule them.